lib/mnesia/test/mnesia_trans_access_test.erl

```diff
@@ -463,8 +467,191 @@ select14(Config) when is_list(Config) ->
+    ?match({atomic, []},
+	   mnesia:transaction(fun() -> mnesia:select_reverse(Tab, OnePat) end)),
+    ?match({atomic, ok},
+	   mnesia:transaction(fun() -> mnesia:write(OneRec) end)),
+    ?match({atomic, ok},
+	   mnesia:transaction(fun() -> mnesia:write(TwoRec) end)),
+    ?match({atomic, [OneRec]},
+	   mnesia:transaction(fun() -> mnesia:select_reverse(Tab, OnePat) end)),
+    %% Need to rpc:call here, as mnesia:ets assumes the ETS table is local to the node
+    ?match([OneRec], rpc:call(Node1, mnesia, ets, [fun() -> mnesia:select_reverse(Tab, OnePat) end])),
+
+    ?match({atomic, [OneRec]}, mnesia:transaction(fun() ->
+	    mnesia:write(ThreeRec),
+	    mnesia:select_reverse(Tab, OnePat)
+    end)),
+
+    ?match({aborted, _},
+	   mnesia:transaction(fun() -> mnesia:select_reverse(Tab, {match, '$1', 2}) end)),
+    ?match({aborted, _},
+	   mnesia:transaction(fun() -> mnesia:select_reverse(Tab, [{'_', [], '$1'}]) end)),
+
+    ?match({'EXIT', {aborted, no_transaction}},  mnesia:select_reverse(Tab, OnePat)),
     ?verify_mnesia(Nodes, []).
 
+%% more select_reverse
+select_reverse14(suite) -> [];
+select_reverse14(Config) when is_list(Config) ->
+    [Node1,Node2] = Nodes = ?acquire_nodes(2, Config),
+    Tab1 = select_reverse14_ets,
+    Tab2 = select_reverse14_dets,
+    Tab3 = select_reverse14_remote,
+    Tab4 = select_reverse14_remote_dets,
+    Tab5 = select_reverse14_xets,
+    Schemas = [[{name, Tab1}, {type, ordered_set}, {attributes, [k, v]}, {ram_copies, [Node1]}],
+	       [{name, Tab2}, {attributes, [k, v]}, {disc_only_copies, [Node1]}],
+	       [{name, Tab3}, {type, ordered_set}, {attributes, [k, v]}, {ram_copies, [Node2]}],
+	       [{name, Tab4}, {attributes, [k, v]}, {disc_only_copies, [Node2]}],
+	       [{name, Tab5}, {type, ordered_set}, {attributes, [k, v]}, {ext_ram_copies, [Node1]}]],
+    [?match({atomic, ok},  mnesia:create_table(Schema)) || Schema <- Schemas],
+
+    %% Some Helpers
+    Trans = fun(Fun) -> mnesia:transaction(Fun) end,
+    Dirty = fun(Fun) -> mnesia:async_dirty(Fun) end,
+    LoopHelp = fun('$end_of_table',_) -> [];
+		  ({Recs,Cont},Fun) ->
+		       Sel = mnesia:select_reverse(Cont),
+		       Recs ++ Fun(Sel, Fun)
+	       end,
+    Loop = fun(Table,Pattern) ->
+		   Sel = mnesia:select_reverse(Table, Pattern, 1, read),
+		   Res = LoopHelp(Sel,LoopHelp),
+		   DescFun = fun(A, B) -> A > B end,
+		   case mnesia:table_info(Table, type) of
+		       ordered_set -> Res;
+		       _ -> lists:sort(DescFun, Res)
+		   end
+	   end,
+    Test =
+	fun(Tab, TabType) ->
+		OneRec = {Tab, 1, 2},
+		TwoRec = {Tab, 2, 3},
+		OnePat = [{{Tab, '$1', 2}, [], ['$_']}],
+		All = [TwoRec,OneRec],
+		AllPat = [{'_', [], ['$_']}],
+
+		?match({atomic, []}, Trans(fun() -> Loop(Tab, OnePat) end)),
+		?match({atomic, ok},       mnesia:transaction(fun() -> mnesia:write(OneRec) end)),
+		?match({atomic, [OneRec]}, mnesia:transaction(fun() -> mnesia:write(TwoRec), Loop(Tab, OnePat) end)),
```

Comment:
Isn't doing Loop(Tab, OnePat) after mnesia:write pointless? You test the Loop one line below this one already.

Reply:
Or wait, it probably is testing the case where you change things and then do select in same transaction.

Reply:
Yeah exactly, it was just the `write` before, and I added the extra `Loop` to cover that case

Reply:
Thanks for explaining :)